packages/remediations/src/steps/reviewSystems.js

```diff
@@ -0,0 +1,160 @@
+import propTypes from 'prop-types';
+import useFieldApi from '@data-driven-forms/react-form-renderer/dist/esm/use-field-api';
+import useFormApi from '@data-driven-forms/react-form-renderer/dist/esm/use-form-api';
+import {
+    Text,
+    TextContent,
+    Stack, StackItem
+} from '@patternfly/react-core';
+import { InventoryTable } from '@redhat-cloud-services/frontend-components/Inventory';
+import { Provider, useDispatch, useSelector } from 'react-redux';
+import promiseMiddleware from 'redux-promise-middleware';
+import { BrowserRouter as Router } from 'react-router-dom';
+import ReducerRegistry from '@redhat-cloud-services/frontend-components-utilities/ReducerRegistry';
+import { ExclamationCircleIcon } from '@patternfly/react-icons';
+import { dedupeArray, EXISTING_PLAYBOOK, inventoryEntitiesReducer as entitiesReducer, TOGGLE_BULK_SELECT } from '../utils';
+import './reviewSystems.scss';
+
+const ReviewSystems = (props) => {
+    const { issues, systems, registry } = props;
+
+    let dispatch = useDispatch();
+    const inventory = useRef(null);
+    const { input } = useFieldApi(props);
+    const formOptions = useFormApi();
+
+    const [ inventoryApi, setInventoryApi ] = useState();
+
+    const { selected, loaded, rows } = useSelector(({ entities }) => ({
+        selected: entities?.selected || [],
+        loaded: entities?.loaded,
+        rows: entities?.rows || []
+    }));
+
+    const formValues = formOptions.getState().values;
+    const error = formOptions.getState().errors?.systems;
+    const playbook = formValues && formValues[EXISTING_PLAYBOOK];
+
+    const allNewSystems = dedupeArray(issues.reduce((acc, curr) => [
+        ...acc,
+        ...(curr.systems || [])
+    ], [ ...systems ]));
+
+    const allSystems = playbook ? dedupeArray(playbook.issues?.reduce((acc, curr) => [
+        ...acc,
+        ...(curr.systems?.map(system => system.id) || [])
+    ], [ ...allNewSystems ])) : allNewSystems;
+
+    const onRefresh = (options, callback) => {
+        if (!callback && inventory && inventory.current) {
+            inventory.current.onRefreshData(options);
+        } else if (callback) {
+            callback(options);
+        }
+    };
+
+    const onSelect = (selected) => {
+        input.onChange(selected);
+    };
+
+    const onSelectRows = (value) => {
+        dispatch({
+            type: TOGGLE_BULK_SELECT,
+            payload: value
+        });
+    };
+
+    const fetchSystemsInfo = useCallback(async (_i, config) => {
+        const systems = (allSystems || []).slice((config.page - 1) * config.per_page, config.page * config.per_page);
+        const data = await inventoryApi?.getEntities(systems, { ...config, hasItems: true, page: 1 }, true);
+        return {
+            ...data,
+            total: allSystems.length,
+            page: config.page,
+            // eslint-disable-next-line camelcase
+            per_page: config.per_page
+        };
+    }, [ allSystems ]);
```

Comment:
Looking at the dependency list it might be better to move this function out of this component
```JS
const fetchSystemsInfo = (config, allSystem, { getEntities } = {}) => {
  // ... rest of the function
}
```